lib/rules/selector-class-pattern/index.mjs

```diff
@@ -123,32 +123,31 @@ const rule = (primary, secondaryOptions) => {
  */
 function hasInterpolatingAmpersand(selector) {
 	for (const [i, char] of Array.from(selector).entries()) {
-		if (char !== '&') {
-			continue;
-		}
-
-		const prevChar = selector.charAt(i - 1);
-
-		if (prevChar && !isCombinator(prevChar)) {
-			return true;
-		}
+		if (char !== '&') continue;
 
-		const nextChar = selector.charAt(i + 1);
-
-		if (nextChar && !isCombinator(nextChar)) {
-			return true;
-		}
+		if (!hasAdjacentCombinator(selector, i)) return true;
 	}
 
 	return false;
 }
 
 /**
- * @param {string} x
- * @returns {boolean}
+ * @param {string} selector
+ * @param {number} index
  */
-function isCombinator(x) {
-	return /[\s+>~]/.test(x);
+function hasAdjacentCombinator(selector, index) {
+	const prevChar = selector.charAt(index - 1);
+	const fn = (/** @type {string} */ character) => /[\s+>~]/.test(character);
```

Comment:
[suggestion] Can we avoid abbreviations and ambiguous function names? The `fn` name makes this code much harder to understand.

Can you instead assign the regular expression to a constant (or moved it to the shared list of regular expressions) and reuse it that way?

Reply:
I would have to change the regular expression if I named it because it would have to support `||` as well.
i.e. `||` is not a single character